bool localize
bool map
bool applyChange
---
bool localize
bool map

